### PR TITLE
Add database view

### LINKED ISSUE #233: Add mockup of tableview of iPads

### PR DESCRIPTION
**Pull request by VegardOztan**
Closes #233, #242, #245, #244, #241, #236

**Changes to:**
- `Order.tsx` and `SimOrderRadio.tsx`
    - Done to make sure 'Order iPad' button in `TablePage.tsx` look OK.
    - ~~Regular Order page now looks a bit wonkier, need to assess this in the future~~
- `Welcome.tsx`
    - Remove topRef
    - Add Database view
- `apiClient.ts`
    - All methods now is now returned the response body and headers
    - Added generic PUT functionality 
    - Added getIpads
    - Added putIpad
 - `models.ts`
   - Added iPad Type and type for DataTable props

**New files:**
- `StatusColumn.tsx`, `SimTypeColumn.tsx` and `ExClassColumn.tsx`
   - Classes used when defining which columns to be shown. Needed as we need to convert the enum to string
- `TablePage.tsx`
   - DataTable showing iPads. Ideally we should have placed 'editIpadEntryButton' (which is the vertical dots menu) in an own file in the components folder, and `DataTableColumn` (which is what defines the columns) in `models.ts`. However, the problem is that by moving 'editIpadEntryButton' to a separate file, it needs to take 'setIsEditOpen', 'setIsChangeOwnerOpen' (and so on...) as inputs, but the 'component' parameter of 'DataTableColumn' wont accept a component which requires input parameters (at least, I can't get it to work).
- `ChangeAssignedPerson.tsx`
   - Handles changing of the assigned person

### CODE REVIEW

File: frontend/src/Order.tsx [hunk 1]
```diff
@@ -17,11 +17,7 @@ import { FieldHeader } from './components/FieldHeader'
 import { WbsPicker } from './components/WbsPicker'
 import { ErrorHandler } from './components/ErrorHandler'
 
-interface Props {
-    topRef: React.RefObject<HTMLElement>
-}
-
-const Order = ({ topRef }: Props) => {
+const Order = () => {
```
Comment (maiaht) [inline frontend/src/Order.tsx:20]: Maybe the Order component can have a prop like `isSideSheet` which defines whether the component is being used in the table view or in the main order view Then the `div`-style and the values for `xs` and `sm` can be set according to the value of `isSideSheet`

File: frontend/src/TablePage.tsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            </ModalSideSheet>
+            <ModalSideSheet
+                header="Change iPad owner"
+                show={isChangeOwnerOpen}
+                onClose={() => {
+                    setIsChangeOwnerOpen(false)
+                }}
+            >
+                <>
+                    Change owner of iPad with id {selectedIpad?.id}. Current owner is {selectedIpad?.owner}
```
[the remark this answers is not in the document]
Comment (VegardOztan) [reply frontend/src/TablePage.tsx:201]: I agree it does, but this was just meant as a placeholder as we don't really know what to put here yet. Maybe we should just put something generic, as `Change owner page` ?

[the remark this answers is not in the document]
Comment (maiaht) [reply frontend/src/TablePage.tsx:201]: Good idea

File: frontend/src/TablePage.tsx [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    const { isFetching, pagination, pagedData, setCurrentPage } = useAsyncPagination(
+        async (pagination: Pagination) => await api.getIpads(pagination.currentPage.index + 1, pageSize),
+        pageSize
+    )
+    const onPaginationChange = useCallback((newPage: Page, perPage: number) => {
+        setCurrentPage(newPage.index, perPage)
+    }, [])
+
+    const editIpadEntryButton: FC<DataItemProps> = ({ item }) => {
```
[the remark this answers is not in the document]
Comment (VegardOztan) [reply frontend/src/TablePage.tsx:43]: Maybe, but I can't get it to work as I outlined in the PR-comment. By moving 'editIpadEntryButton' to a separate file, it needs to take 'setIsEditOpen', 'setIsChangeOwnerOpen' (and so on...) as inputs, but the 'component' parameter of 'DataTableColumn' (see line 165) wont accept a component which requires input parameters.

File: frontend/src/Order.tsx (hunk 1 above)
Comment (VegardOztan) [reply frontend/src/Order.tsx:20]: Yes good idea, will implement

File: frontend/src/TablePage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (maiaht) [reply frontend/src/TablePage.tsx:43]: ok

File: frontend/src/Order.tsx (hunk 1 above)
Comment (VegardOztan) [reply frontend/src/Order.tsx:20]: Fixed

**Review by maiaht — APPROVED** [review]

**Pull request merged**